lib/cli/src/commands/package/tag.rs

```diff
@@ -12,14 +12,18 @@ use is_terminal::IsTerminal;
 use std::{
     path::{Path, PathBuf},
     str::FromStr,
+    sync::OnceLock,
 };
-use wasmer_api::WasmerClient;
+use wasmer_api::{types::PackageVersionWithPackage, WasmerClient};
 use wasmer_config::package::{
     Manifest, NamedPackageId, NamedPackageIdent, PackageBuilder, PackageHash, PackageIdent,
 };
 
 use super::PublishWait;
 
+/// The package related to the NamedPackageId the user chose to tag.
+static MAYBE_USER_PACKAGE: OnceLock<Option<PackageVersionWithPackage>> = OnceLock::new();
```

Comment:
A shared global for this is quite nasty... 

Why is this neccessary?

Reply:
In order to avoid repeating the same query to the backend twice while keeping function types a bit cleaner. Otherwise, the `get_version` function should return a `Result<(Option<PackageVersionWithPackage>, semver::Version)>` lifting up to `synthesize_id`. To me - this is of course subjective - is less clean than a `OnceLock`.